common-npm-packages/AzureRmDeploy-common/azure-arm-rest/azure-arm-app-service.ts

```diff
@@ -263,6 +263,13 @@ export class AzureAppService {
                 tl.debug(`Slot setting updated for key : ${addProperties[key].name}`);
                 isNewValueUpdated = true;
             }
+            else if (addProperties[key].slotSetting == false || (addProperties[key].slotSetting == null)) {
+                const index = appSettingNames.indexOf(addProperties[key].name, 0);
+                if (index > -1) {
+                    appSettingNames.splice(index, 1);
+                }
```

Comment:
Can you run this test matrix and check if all is working correctly ?
https://docs.google.com/spreadsheets/d/144O20MF0IbC-O9vW-8wSBBUEpIUtMIf56CAlCO194jI/edit?usp=sharing